src/java.base/share/classes/java/util/Optional.java

```diff
@@ -394,7 +394,7 @@ public T orElseThrow() {
      * @return the value, if present
      * @throws X if no value is present
      * @throws NullPointerException if no value is present and the exception
-     *          supplying function is {@code null}
+     *         supplying function or produces a {@code null} result
```

Comment:
It seems there is a word missing before the "or".
```suggestion
     *         supplying function is {@code null} or produces a {@code null} result
```